modin/core/execution/dask/implementations/pandas_on_dask/partitioning/partition.py

```diff
@@ -62,10 +63,9 @@ def get(self):
             The object from the distributed memory.
         """
         self.drain_call_queue()
-        # blocking operation
-        if isinstance(self.future, pandas.DataFrame):
-            return self.future
-        return self.future.result()
+        res = self.future.result()
+        assert isinstance(res, pandas.DataFrame)
```

Comment:
@YarShev this check cannot be used here, since after calling `apply` function, an arbitrary object can be in the partition (for example - `numpy` object).

```bash
modin/core/dataframe/pandas/partitioning/partition.py:147: in to_numpy
    return self.apply(lambda df, **kwargs: df.to_numpy(**kwargs)).get()
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <modin.core.execution.ray.implementations.pandas_on_ray.partitioning.partition.PandasOnRayDataframePartition object at 0x7f8208415700>

    def get(self):
        """
        Get the object wrapped by this partition out of the Plasma store.
    
        Returns
        -------
        pandas.DataFrame
            The object from the Plasma store.
        """
        self.drain_call_queue()
        res = ray.get(self.oid)
>       assert isinstance(res, pandas.DataFrame)
E       AssertionError
```